src/workerd/api/web-socket.h

```diff
@@ -497,6 +526,22 @@ class WebSocket: public EventTarget {
       // If we are "releasing", we may prevent the websocket from doing certain things like calling
       // send/close. We're more restrictive if we're delivering an Error than delivering a Close.
       HibernatableReleaseState releaseState = HibernatableReleaseState::NONE;
+
+    // There are two possible states for tagsRef:
+    //  1. kj::Function
+    //          - A function pointer that lets us call HibernatableWebSocket::getTags()
+    //            from api::WebSocket.
+    //          - So long as the HibernatableWebSocket is around, we can just use call its method.
+    //  2. kj::Array<kj::String>
+    //          - An array of the tags.
+    //          - We're going to be dispatching a Close or an Error event, i.e. the
+    //            HibernatableWebSocket is free to go away. We can no longer rely on its getTags()
+    //            method, so instead we move the data into the api::WebSocket.
+    //
+    // We could just copy all tags into api::WebSocket everytime we reactivate/wake from
+    // hibernation, but it could add up to 2.56KB of memory for each websocket.
+    // With a maximum of 32k websockets, that could put a lot of memory pressure on the DO.
+      kj::OneOf<kj::Function<kj::Array<kj::String>()>, kj::Array<kj::String>> tagsRef;
```

Comment:
How about we just make this a function? We can always make a `kj::Function` that just returns an array/copy for when we're closing instead of calling `getTags()` and it will probably make the code easier to follow.